lib/diagnostic.js

```diff
@@ -5,17 +5,29 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function ( previous, current )
+{
+  let result = array[0];
+  let minimum = function ( array, callback )
+  {
+    return array.forEach( e, i, a )
+    {
+      if( result[i] > array[i] )
+      {
+        resut[i] = array[i];
+      }
+    }
+  };
 };
```

Comment:
You've misunderstood the question here. `min` is supposed to be a _callback_. Do you remember what a callback is? This means that `min` will be called by another function. Take a look at this solution:
```
const min = function (previous, current) {
  return previous < current ? previous : current;
};
```